images/multi-platform/Dockerfile

```diff
@@ -4,23 +4,44 @@
 #
 
 FROM --platform=$BUILDPLATFORM docker.io/library/golang:1.24.3 AS builder
-
-WORKDIR /src
-COPY . .
-ENV CGO_ENABLED=0
-
+# Build arguments used for cross-compiling
 ARG TARGETOS
 ARG TARGETARCH
 ENV GOOS=${TARGETOS}
 ENV GOARCH=${TARGETARCH}
+ENV CGO_ENABLED=0 \
+    GO111MODULE=on
+WORKDIR /src
 
+# Copy only dependency files first (better layer caching)
+COPY go.mod go.sum ./
+RUN go mod download
```

Comment:
I am not sure we need that. Arma is using vendoring, so all the dependencies will be copied in the next stage  (`COPY . .`) anyway right into the `vendor` directory.